pkg/cli/admin/sosreport/sosreport.go

```diff
@@ -0,0 +1,687 @@
+package sosreport
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	imagev1client "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	imagereference "github.com/openshift/library-go/pkg/image/reference"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+	"github.com/openshift/oc/pkg/cli/admin/inspect"
+	"github.com/openshift/oc/pkg/cli/rsync"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubectl/pkg/cmd/logs"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/polymorphichelpers"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	sosReportLong = templates.LongDesc(`
+		Launch a pod to collect a SOSReport from the provided node-name
+
+		This command will launch a pod in a temporary namespace on your cluster that gathers
+		a SOSReport from the provided node and then downloads the collected report.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	sosReportExample = templates.Examples(`
+		# Collect a SOSReport using the default image configuration, writing the SOSReport into ./<host-name>.sosreport
+		  oc adm sos
+	`)
+)
+
+func NewSOSReportCommand(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
```

Comment:
@TurboTurtle please review! 
I don't think we want to be calling sosreport directly on N nodes; but instead call `sos collect` in these situations; to pull data from N nodes. Is that correct?

Reply:
Correct - the preferred way (from an `sos` perspective anyways) would be to run `sos collect` on a single node.

There are 2 approaches here:

1. Run `sos collect` bare on a node that can return a list of nodes via `oc get nodes`. This can either be done by configuring the node such that `root` can run `oc get nodes` successfully, or by passing an access token to `sos collect` that allows `oc get nodes` to be successful. This token can either be passed via the `-c ocp.token=<token>` option, or could be set in the `SOSOCPTOKEN` env var, the latter of which prevents the token from appearing in process listings.

2. Run `sos collect --nodes=node1,node2,etc...` where the calling image directly provides the nodes to collect from.